app/controllers/analytics_controller.rb

```diff
@@ -33,99 +33,51 @@ def enroller_stats
+    dates.each do |date|
+      next if date.nil?
 
-    # Symptomatic or unsymptomatic per day
-    symptomatic_assessments_by_day = Hash.new(0)
-    asymptomatic_assessments_by_day = Hash.new(0)
-    Assessment.where(patient: patients).find_each do |a|
-      if a.symptomatic
-        symptomatic_assessments_by_day[a.created_at.to_date] += 1
-      else
-        asymptomatic_assessments_by_day[a.created_at.to_date] += 1
-      end
-    end
-    assessment_result_by_day_array = []
-    (symptomatic_assessments_by_day.keys | asymptomatic_assessments_by_day.keys).sort.each do |date|
+      # Get last saved analytic for each date
+      analytic = jurisdiction_analytics.where(created_at: date.beginning_of_day..date.end_of_day).last
+      root_analytic = root_jurisdiction_analytics.where(created_at: date.beginning_of_day..date.end_of_day).last
+      open_cases = !analytic&.open_cases_count.nil? ? analytic.open_cases_count : 0
+      patient_count_by_day_array << { day: date, cases: open_cases }
+      symp_count = !analytic&.symptomatic_monitorees_count.nil? ? analytic.symptomatic_monitorees_count : 0
       assessment_result_by_day_array << {
         'name' => date.to_s,
-        'Symptomatic Assessments' => symptomatic_assessments_by_day[date],
-        'Asymptomatic Assessments' => asymptomatic_assessments_by_day[date]
+        'Symptomatic Assessments' => symp_count,
+        'Asymptomatic Assessments' => open_cases - symp_count
       }
+      # Map analytics are pulled from the root (most likely USA) jurisdiction
+      sym_map = root_analytic&.monitoree_state_map.nil? ? (JSON.parse root_analytic.monitoree_state_map.gsub('=>', ':')) : {}
+      symptomatic_patient_count_by_state_and_day << { day: date }.merge(sym_map)
+      count_map = !root_analytic&.symptomatic_state_map.nil? ? (JSON.parse root_analytic.symptomatic_state_map.gsub('=>', ':')) : {}
+      total_patient_count_by_state_and_day << { day: date }.merge(count_map)
     end
 
+    most_recent_analytics = current_user.jurisdiction.analytics.last
+
     {
-      system_subjects: patients.count,
-      system_subjects_last_24: patients.where('created_at >= ?', Time.now - 1.day).count,
-      system_assessments: Assessment.where(patient: patients).count,
-      system_assessments_last_24: Assessment.where(patient: patients).where('created_at >= ?', Time.now - 1.day).count,
+      last_updated_at: most_recent_analytics.updated_at,
       subject_status: [
-        { name: 'Asymptomatic', value: asymptomatic_patients.length },
-        { name: 'Non-Reporting', value: non_reporting_patients.length },
-        { name: 'Symptomatic', value: symptomatic_patients.length },
-        { name: 'Closed', value: closed_patients.count }
+        # TODO: Determine if asymptomatic included closed patients too
+        { name: 'Asymptomatic', value: most_recent_analytics.open_cases_count - most_recent_analytics.symptomatic_monitorees_count },
```

Comment:
There is a `asymptomatic` scope already, why not use that (add a asymptomatic field to the model, make the rake task populate it, us it here)? This is incorrect at the moment because I think it is still including non-reporting.

Reply:
I was trying to be clever by reducing the number of queries that the analytic needed to do in order to get the information it needed and reduce the size of the analytic entry. It shouldn't really add that much time/memory so i'll just add a field that maps directly to patients.asymptomatic the same way symptomatic was done